### PR TITLE
Prototype volunteer registration

### PR DESCRIPTION
**Pull request by paul-tennyson**

### CODE REVIEW

File: BPOR/BPOR.Rms/Controllers/VolunteerController.cs [hunk 1]
```diff
@@ -11,6 +12,123 @@ namespace BPOR.Rms.Controllers;
...
+
+        if (ModelState.IsValid)
+        {
+            return RedirectToAction(nameof(VolunteerController.AccountSuccess));
+        }
+
+        return View(model);
+    }
+
+    private void DoesPostcodeSurnameDoBComboExist(string postCode, string lastName, GovUkDate dateOfBirth)
```
Comment (petecorriepa) [inline BPOR/BPOR.Rms/Controllers/VolunteerController.cs:82]: this method can be marked as async and return a Task
then use await (line 86).... FirstOrDefaultAsync() (line 90)

File: BPOR/BPOR.Rms/Controllers/VolunteerController.cs [hunk 2]
```diff
@@ -11,6 +12,123 @@ namespace BPOR.Rms.Controllers;
...
+        p.Address.Postcode == postCode)
+        .FirstOrDefault();
+
+        if (user != null)
+        {
+            ModelState.AddModelError("LastName", "Combination of surname, date of birth and postcode already exists and cannot be used");
+        }
+    }
+
+    private void DoesUserEmailExistInDatabase(string email)
```
Comment (petecorriepa) [inline BPOR/BPOR.Rms/Controllers/VolunteerController.cs:98]: same here, this can be async

File: BPOR/BPOR.Rms/Controllers/VolunteerController.cs [hunk 3]
```diff
@@ -11,6 +12,123 @@ namespace BPOR.Rms.Controllers;
...
+    private void ValidateDateOfBirth(GovUkDate dateOfBirth)
+    {
+        if (!dateOfBirth.HasValue)
+        {
+            ModelState.AddModelError("DateOfBirth", "Date of birth is required");
+        }
+
+        if (dateOfBirth.HasValue)
+        {
+            DateOnly today = DateOnly.FromDateTime(DateTime.Today);
```
Comment (petecorriepa) [inline BPOR/BPOR.Rms/Controllers/VolunteerController.cs:117]: should this be >=?
I can sign up on my 18th birthday

File: BPOR/BPOR.Rms/Controllers/VolunteerController.cs (hunk 1 above)
Comment (paul-tennyson) [reply BPOR/BPOR.Rms/Controllers/VolunteerController.cs]: Thanks @petecorriepa I've updated these now to use async tasks

File: BPOR/BPOR.Rms/Controllers/VolunteerController.cs (hunk 3 above)
Comment (paul-tennyson) [reply BPOR/BPOR.Rms/Controllers/VolunteerController.cs:121]: The current method actually does accept a date that would on on your 18th birthday. If I input 18 09 2006, it accepts this. If I input 19 09 2006, it flags as a fail.

Comment (petecorriepa) [reply BPOR/BPOR.Rms/Controllers/VolunteerController.cs:121]: ok my bad!

**Review by petecorriepa — APPROVED** [review]

**Pull request closed without merging**